solrsearch-plugins/src/main/java/co.cask.hydrator.plugin/common/SolrSearchSinkConfig.java

```diff
@@ -0,0 +1,221 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package co.cask.hydrator.plugin.common;
+
+import co.cask.cdap.api.annotation.Description;
+import co.cask.cdap.api.data.schema.Schema;
+import co.cask.hydrator.common.ReferencePluginConfig;
+import com.google.common.base.Splitter;
+import org.apache.commons.lang.StringUtils;
+import org.apache.solr.client.solrj.SolrClient;
+import org.apache.solr.client.solrj.impl.CloudSolrClient;
+import org.apache.solr.client.solrj.impl.HttpSolrClient;
+
+import java.util.HashMap;
+import java.util.Map;
+import javax.annotation.Nullable;
+
+/**
+ * Config class for SolrSearchSink and RealtimeSolrSearchSink plugins.
+ */
+public class SolrSearchSinkConfig extends ReferencePluginConfig {
+  public static final String SINGLE_NODE_MODE = "SingleNode";
+  public static final String SOLR_CLOUD_MODE = "SolrCloud";
+
+  @Description("Solr mode to connect to. For example, Single Node Solr or SolrCloud.")
+  private final String solrMode;
+  @Description("The hostname and port for the Solr server separated by a colon. For example, localhost:8983 for " +
+    "Single Node Solr or comma-separated list of hostname and port, zkHost1:2181,zkHost2:2181,zkHost3:2181 " +
+    "for SolrCloud.")
+  private final String solrHost;
+  @Description("Name of the collection where data will be indexed and stored in Solr.")
+  private final String collectionName;
+  @Description("Field that will determine the unique key for the document to be indexed. It must match a " +
+    "field name in the structured record of the input.")
+  private final String keyField;
+  @Description("List of the input fields to map to the output Solr fields. The key specifies the name of the field to" +
+    " rename, with its corresponding value specifying the new name for that field.")
+  @Nullable
+  private final String outputFieldMappings;
+
```

Comment:
if some field is used to determine schema and validate, then it makes sense to not make them macro enabled.